src/weapon/w_024.c

```diff
@@ -181,7 +181,111 @@ int GetWeaponId(void) { return 24; }
+        self->flags |= FLAG_UNK_04000000 | FLAG_HAS_PRIMS;
+        self->ext.shield.unk9A = rand() & 0xF;
+        self->ext.shield.unk94 = self->posX.i.hi = (rand() % 248) + 4;
+        self->posY.i.hi = (rand() & 0x3F) + 0xD8;
+        self->facingLeft = rand() & 1;
+
+        prim = &g_PrimBuf[self->primIndex];
+        prim->tpage = 0x19;
+        prim->clut = self->ext.shield.childPalette;
+        if (self->ext.shield.unk9A < 8) {
+            prim->u0 = prim->u2 = 0x50;
+            prim->u1 = prim->u3 = 0x50 + 15;
+            prim->v0 = prim->v1 = self->ext.shield.unk7D;
+            prim->v2 = prim->v3 = self->ext.shield.unk7D + 0x1F;
+            self->ext.shield.unk84 = ((self->ext.shield.unk9A + 8) * 8) / 10;
+            self->ext.shield.unk86 = ((self->ext.shield.unk9A + 8) * 0x10) / 10;
+            prim->r2 = prim->g2 = prim->b2 = prim->r3 = prim->g3 = prim->b3 =
+                -0x80 - ((7 - self->ext.shield.unk9A) * 0xF);
+            prim->priority = 0x1B8;
+        } else {
+            prim->u0 = prim->u2 = 0x60;
+            prim->u1 = prim->u3 = 0x60 + 31;
+            prim->v0 = prim->v1 = self->ext.shield.unk7D;
+            prim->v2 = prim->v3 = self->ext.shield.unk7D + 0x2F;
+            self->ext.shield.unk84 = (self->ext.shield.unk9A * 0x10) / 10;
+            self->ext.shield.unk86 = (self->ext.shield.unk9A * 0x18) / 10;
+            prim->priority = 0x1BE;
+            prim->r2 = prim->g2 = prim->b2 = prim->r3 = prim->g3 = prim->b3 =
+                -0x80 - ((0xF - self->ext.shield.unk9A) * 0xC);
+        }
+        prim->r1 = prim->g1 = prim->b1 = prim->r0 = prim->g0 = prim->b0 = 0x80;
+        prim->drawMode = DRAW_COLORS;
+        self->velocityY = -((self->ext.shield.unk9A * FIX(3.0 / 16)) + FIX(3));
+        self->ext.shield.unkA0 = -((self->ext.shield.unk9A << 8) + 0x900);
+        self->ext.shield.unkAE = self->ext.shield.parent->ext.shield.unkAE;
+        SetWeaponProperties(self, 0);
+        self->hitboxHeight = 0x40;
+        self->hitboxWidth = 0x40;
+        self->step += 1;
+        break;
+    case 1:
+        self->posY.val += self->velocityY / 16;
+        if (self->ext.shield.parent->step > 5) {
+            self->ext.shield.unk80 = (rand() % 48) + 10;
+            self->step += 1;
+        }
+        break;
+    case 2:
+        self->posX.i.hi = self->ext.shield.unk94 + (self->ext.shield.unk80 & 1);
+        self->posY.val += self->velocityY / 32;
+        if (--self->ext.shield.unk80 == 0) {
+            self->velocityY /= 32;
+            g_api.PlaySfx(SFX_DARK_SHIELD);
+            self->step += 1;
+        }
+        break;
+    case 3:
+        self->posX.i.hi = self->ext.shield.unk94 + (self->ext.shield.unk80 & 1);
+        self->posY.val += self->velocityY;
+        self->velocityY += self->ext.shield.unkA0;
+        if (self->posY.i.hi < 0x80) {
+            self->ext.shield.unk92 += 2;
+            self->ext.shield.unkA0 -= 0x2000;
```

Comment:
This is added to velocityY so I think it's FIX?

Reply:
Good call, thanks